domain_model/Election.ts

```diff
@@ -39,6 +40,27 @@ export function electionValidation(obj:Election): string | null {
   return null;
 }
 
-export function removeHiddenFields(obj:Election):void {
+export function removeHiddenFields(obj:Election, electionRoll: ElectionRoll|null):void {
   obj.auth_key = undefined;
+  if (obj.state==='open' && electionRoll?.precinct){
+    // If election is open and precinct is defined, remove races that don't include precinct
+    obj.races = getApprovedRaces(obj, electionRoll.precinct)
+    
+  }
+}
+// Where should this belong..
+export function getApprovedRaces(election:Election, precinct:string|undefined) {
```

Comment:
nit:  specify the argument is meant to be the voter's precinct.  type the output as list of Races